wicket-core/src/main/java/org/apache/wicket/markup/html/form/upload/OnFilesSelectedBehavior.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+;(function (undefined) {
+
+    'use strict';
+
+    if (typeof(Wicket.OnFilesSelected) === 'object') {
+        return;
+    }
+
+    /**
+     * Contains JS code for OnFilesSelectedBehavior.
+     */
+    Wicket.OnFilesSelected = {
+
+        // the id of temporary object holding files.
+        varId: function (componentId) {
+            return componentId + "_files";
+        },
+
+        /**
+         * Precondition to trigger (oor not) server round-trip.
+         *
+         * @param component
+         * @param componentId
+         * @returns {boolean} true if some files (file) were (was) selected,
+         */
+        precondition : function (component, componentId) {
+            var id = this.varId(componentId);
+            if (component.files && component.files.length > 0) {
+                window[id] = component.files;
```

Comment:
Can we use `Wicket.OnFilesSelected[id]` instead of `window[id]` ?
We should try to not pollute the global namespace, even temporarily.

Reply:
Yes. I think we can.

Reply:
fixed